src/Ads/AssetSuggestionsService.php

```diff
@@ -0,0 +1,211 @@
+			]
+		);
+
+		foreach ( $terms as $term ) {
+				$terms_suggestions[] = $this->format_final_url_response( $term->term_id, 'term', $term->name, get_term_link( $term->term_id, $term->taxonomy ) );
+		}
+
+		return $terms_suggestions;
+
+	}
+
+
+	/**
+	 * Return a list of final urls that can be used to suggest assets.
+	 *
+	 * @param string $search The search query
+	 * @param int    $per_page Number of items per page
+	 * @param string $order_by Order by: id, type, title, url
+	 *
+	 * @return array final urls with their title, id & type.
+	 */
+	public function get_final_urls_suggestions( $search = '', $per_page = 30, $order_by = 'title' ): array {
+		if ( empty( $search ) ) {
+			 return $this->get_defaults_final_urls_suggestions();
+		}
+
+		// Split possible results between posts and terms.
+		$per_page_posts = (int) ceil( $per_page / 2 );
+
+		$posts = $this->get_post_suggestions( $search, $per_page_posts );
+
+		// Try to get more results using the terms
+		$per_page_terms = $per_page - count( $posts );
+
+		$terms = [];
+
+		// get_terms  evaluates $per_page_terms = 0 as a falsy, therefore it will not add the LIMIT clausure returning all the results.
+		// See: https://github.com/WordPress/WordPress/blob/abe134c2090e84080adc46187884201a4badd649/wp-includes/class-wp-term-query.php#L868
+		if ( $per_page_terms > 0 ) {
+			$terms = $this->get_terms_suggestion( $search, $per_page_terms );
+		}
+
+		$pending_results = $per_page - count( $posts ) - count( $terms );
+		$more_results    = [];
+
+		// Try to get more results using posts
+		if ( $pending_results > 0 && count( $posts ) === $per_page_posts ) {
+			$more_results = $this->get_post_suggestions( $search, $pending_results, $per_page_posts );
+		}
+
+		$result = array_merge( $posts, $terms, $more_results );
+
+		return $this->sort_results( $result, $order_by );
+
+	}
+
+	/**
+	 * Get defaults final urls suggestions.
+	 *
+	 * @return array default final urls.
+	 */
+	protected function get_defaults_final_urls_suggestions(): array {
+		// We can only offer assets if the homepage is static.
+		$home_page = $this->wp->get_static_homepage();
+		$shop_page = $this->wp->get_shop_page();
+		$defaults  = [];
+
+		if ( $home_page ) {
+			$defaults[] = $this->format_final_url_response( $home_page->ID, 'post', 'Homepage', get_permalink( $home_page->ID ) );
+		}
+
+		if ( $shop_page ) {
+			$defaults[] = $this->format_final_url_response( $shop_page->ID, 'post', $shop_page->post_title, get_permalink( $shop_page->ID ) );
+		}
+
+		return $defaults;
+	}
+
+	/**
+	 *  Order suggestions alphabetically
+	 *
+	 *  @param array  $array associative array
+	 *  @param string $field Sort by a specific field
+	 *
+	 * @return array response sorted alphabetically
+	 */
+	protected function sort_results( $array, $field ): array {
+		usort(
+			$array,
+			function ( $a, $b ) use ( $field ) {
+				return strcmp( (string) $a[ $field ], (string) $b[ $field ] );
+			}
+		);
+
+		return $array;
+
+	}
+
```

Comment:
This is not working for ID `order_by`. . Unsure how essential is to order by id. Maybe it is just ok to remove that `order_by=id` instead of adding complexity in the sort function. 

Another thing to consider is that this will place lowercase after uppercase. See last entries of my response ordering by Title. Maybe we could lowercase the fields.

```
{
        "id": 182,
        "type": "post",
        "title": "Term Post",
        "url": "https://wp23.test/term-post/"
    },
    {
        "id": 35,
        "type": "post",
        "title": "WordPress Pennant",
        "url": "https://wp23.test/product/wordpress-pennant/"
    },
    {
        "id": 166,
        "type": "post",
        "title": "p4", // starts with p, but appears after W.
        "url": "https://wp23.test/p4/"
    }
```

Reply:
I excluded the order by id as it is not essential. I think ordering by title will help the user to find the desired result. See [a3c8b5a](https://github.com/woocommerce/google-listings-and-ads/pull/1784/commits/a3c8b5a9fe9eb5d9a2c298590cf81dae836816af)